x/dex/keeper/cancel_limit_order.go

```diff
@@ -88,8 +88,8 @@ func (k Keeper) ExecuteCancelLimitOrder(
 		return sdk.Coin{}, sdk.Coin{}, nil, sdkerrors.Wrapf(types.ErrCancelEmptyLimitOrder, "%s", tranche.Key.TrancheKey)
 	}
 
-	k.SaveTrancheUser(ctx, trancheUser)
-	k.SaveTranche(ctx, tranche)
+	k.SaveOrRemoveTrancheUser(ctx, trancheUser)
+	k.SaveOrMakeInactiveTranche(ctx, tranche)
```

Comment:
according to the [docs](https://docs.neutron.org/neutron/modules/dex/overview/concepts/tick-liquidity), we set limit order tranches inactive in begin blockers based on their expiration time. why there is a chance we set to inactive here? is this documentation part up to date?

<img width="1244" alt="image" src="https://github.com/user-attachments/assets/5993bf79-5289-4174-8c13-21e408a655e3">